storage/badger/operation/transaction_results.go

```diff
@@ -46,3 +46,26 @@ func LookupTransactionResultsByBlockID(blockID flow.Identifier, txResults *[]flo
 
 	return traverse(makePrefix(codeTransactionResult, blockID), txErrIterFunc)
 }
+
+// LookupTransactionResultsByBlockIDUsingIndex retrieves all tx results for a block, but using
+// tx_index index. This correctly handles cases of duplicate transactions within block, and should
+// eventually replace uses of LookupTransactionResultsByBlockID
+func LookupTransactionResultsByBlockIDUsingIndex(blockID flow.Identifier, txResults *[]flow.TransactionResult) func(*badger.Txn) error {
+
+	txErrIterFunc := func() (checkFunc, createFunc, handleFunc) {
```

Comment:
Because its named like this in a function above. I guess it should be `txResultsIterFunc` for both maybe? Or its intended to iterate on transaction errors

Reply:
seems like it should be `txResultsIterFunc`?